app/models/user.rb

```diff
@@ -0,0 +1,16 @@
+class User 
+  attr_reader :hash
+  
+  def initialize(session)
+    session[:selection_owner_hash] ||= User.generate_owner_hash
```

Comment:
I would probably move this line to `find_session` and just pass the secure random hash into the `User.new` method. Which leads me to a next comment: In cases like this, it's tricky to be clear when using the word "hash". You could mean a string that is a one-way encrypted hash of some input, or you could mean a collection of key/values. It might be nice to change `@hash` and the `attr_reader` to be `id`, which is more about what it is (a unique identifier) than how it is represented.